File: Extensions/Sixeyed.Extensions.Advanced/Sixeyed.Extensions.Advanced.Tests/Demo4/StringExtensionsShould.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sixeyed.Extensions.Advanced.Tests.Demo4
{
    [TestClass]
    public class StringExtensionsShould
    {
        [TestMethod]
        public void ReturnAFormattedString_GivenAString_WhenTestingFormatWith()
        {
            var actual = "Demo {0}".FormatWith(4);
            var expected = "Demo 4";

            Assert.AreEqual(expected, actual);
        }
    }
}

File: Extensions/Sixeyed.Extensions.Advanced/Sixeyed.Extensions.Advanced.VbTests/Demo4/StringExtensionsShould.vb
Imports System.Text
Imports System
Imports Microsoft.VisualStudio.TestTools.UnitTesting

<TestClass()> Public Class StringExtensionsShould

    <TestMethod()> Public Sub ReturnDemo4_GivenStringDemoAnd4_WhenTestingFormatWith()

        Dim expected As String = "Demo 4"
        Dim actual As String = "Demo {0}".FormatWith(4)

        Assert.AreEqual(expected, actual)
    End Sub

End Class